src/app/get-started/sample-card/sample-card.html

```diff
@@ -14,6 +14,9 @@
 <div class="pf-c-card pf-m-hoverable pf-m-compact pf-m-selectable get-started-template" tabindex="-1">
   <div class="pf-c-card__head">
     <img ng-if="devfile.icon" src="{{devfile.icon}}"/>
+    <div ng-if="!devfile.icon" class="blank-icon">
```

Comment:
I think the better variant to create a needed blob for catch method:
https://github.com/eclipse/che-dashboard/blob/98d3769ab123636dbf935a8e47eb7beecde9d228/src/components/attribute/img-src/img-src.directive.ts#L63
and remove all <div class="chefont cheico-type-blank"> unnecessary conditions.

WDYT?

Reply:
Conditions are used here to define whether to show a devfile icon or the default blank icon. I don't see how I can get rid of these `ng-if`s. Could you elaborate on this?

Reply:
Get the content from the target icon here: 'src/assets/svg/type-blank.svg'
Converting it with a btoa and use it as 
src="data:image/svg+xml;base64, PCFET0NUWVBFIHN2ZyBQVUJMSUMgIi0vL1czQy8vRFREIFNWRyAxLjEvL0VOIiAiaHR0cDovL3d3dy53My5vcmcvR3JhcGhpY3MvU1ZHLzEuMS9EVEQvc3ZnMTEuZHRkIj4KPHN2ZyB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHg9IjBweCIgeT0iMHB4IgoJIHdpZHRoPSI1MTJweCIgaGVpZ2h0PSI1MTJweCIgdmlld0JveD0iMCAwIDUxMiA1MTIiIHN0eWxlPSJlbmFibGUtYmFja2dyb3VuZDpuZXcgMCAwIDUxMiA1MTI7IiB4bWw6c3BhY2U9InByZXNlcnZlIj4KPHBhdGggaWQ9Imljb24iIGQ9Ik02Ny4yMzMsMTUyLjk1OEwyNTYsNTBsMTg4LjY2NiwxMDQuMDIzTDI1Ni4wODMsMjU0LjYzNEw2Ny4yMzMsMTUyLjk1OHogTTI3My4xNjcsMjg0LjczNFY0NjIKCWwxNzEuNjY2LTkyLjAzNlYxOTIuNjgxTDI3My4xNjcsMjg0LjczNHogTTIzOC44MzMsMjg0LjczNGwtMTcxLjY2Ni05Mi45djE3OC4xMjlMMjM4LjgzMyw0NjJWMjg0LjczNHoiLz4KPC9zdmc+"

```
      this.$http<string>(requestConfig).then((response: ng.IHttpResponse<string>) => {
        const blob = new Blob([response.data], {type: response.headers('Content-Type')});
        $attrs.$set('src', (window.URL || (window as any).webkitURL).createObjectURL(blob));
        onload(true);
      }, () => {
        onload(false);
      });
```
Change to
```
      this.$http<string>(requestConfig).then((response: ng.IHttpResponse<string>) => {
        const blob = new Blob([response.data], {type: response.headers('Content-Type')});
        $attrs.$set('src', (window.URL || (window as any).webkitURL).createObjectURL(blob));
        onload(true);
      }, () => {
        $attrs.$set('src', "data:image/svg+xml;base64, `PCFET0NUWVBFIHN2ZyBQVUJMSUMgIi0vL1czQy8vRFREIFNWRyAxLjEvL0VOIiAiaHR0cDovL3d3dy53My5vcmcvR3JhcGhpY3MvU1ZHLzEuMS9EVEQvc3ZnMTEuZHRkIj4KPHN2ZyB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHg9IjBweCIgeT0iMHB4IgoJIHdpZHRoPSI1MTJweCIgaGVpZ2h0PSI1MTJweCIgdmlld0JveD0iMCAwIDUxMiA1MTIiIHN0eWxlPSJlbmFibGUtYmFja2dyb3VuZDpuZXcgMCAwIDUxMiA1MTI7IiB4bWw6c3BhY2U9InByZXNlcnZlIj4KPHBhdGggaWQ9Imljb24iIGQ9Ik02Ny4yMzMsMTUyLjk1OEwyNTYsNTBsMTg4LjY2NiwxMDQuMDIzTDI1Ni4wODMsMjU0LjYzNEw2Ny4yMzMsMTUyLjk1OHogTTI3My4xNjcsMjg0LjczNFY0NjIKCWwxNzEuNjY2LTkyLjAzNlYxOTIuNjgxTDI3My4xNjcsMjg0LjczNHogTTIzOC44MzMsMjg0LjczNGwtMTcxLjY2Ni05Mi45djE3OC4xMjlMMjM4LjgzMyw0NjJWMjg0LjczNHoiLz4KPC9zdmc+`);
        onload(true);
      });
```

![Screenshot from 2020-03-25 13-11-46](https://user-images.githubusercontent.com/6310786/77531360-74fdd580-6e9b-11ea-8ddb-708a9c3a06ce.png)

Reply:
Thanks for explanation, I got your point. But I don't see a profit here, it only makes things implicit. BTW, directive `img-src` was introduced for doing one particular thing which is requesting image source when authorization header is required.

Reply:
Ok. It's Up To You